src/base/bittorrent/session.h

```diff
@@ -314,6 +314,10 @@ namespace BitTorrent
         void setForceProxyEnabled(bool enabled);
         bool isProxyPeerConnectionsEnabled() const;
         void setProxyPeerConnectionsEnabled(bool enabled);
+        bool chokingAlgorithm() const;
```

Comment:
Why do you provide originally enumerable setting as bool?
The same question is about mixed mode algorithm.